src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,129 @@
+package duke.ui;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * Ui create messages that are being display to user.
+ */
+public class Ui {
+
+    /**
+     * Returns exit message when user exit the programs.
+     *
+     * @return Exit message when user exit the programs.
+     */
+    public String showExit() {
+        String bye = "Bye. Hope to see you again soon!";
+        return bye;
+    }
+
+    /**
+     * Returns task added to the list.
+     *
+     * @param task Task that is added.
+     * @param taskList The list of tasks.
+     * @return Task added to the list.
+     */
+    public String showAddTask(Task task, TaskList taskList) {
+        String addTask = String.format("Got it. I've added this task:\n%s\nNow you have %d task(s) in the list.",
+                task.toString(), taskList.totalTask());
+        return addTask;
+    }
+
+    /**
+     * Returns task that is deleted from the list.
+     *
+     * @param task Task that is deleted.
+     * @param taskList The list of tasks.
+     * @return Task that is deleted from the list.
+     */
+    public String showDeleteTask(Task task, TaskList taskList) {
+        String deleteTask = String.format("Noted. I've removed this task:\n%s\nNow you have %d task(s) in the list.",
+                task.toString(), taskList.totalTask());
+        return deleteTask;
+    }
+
+    /**
+     * Returns task that is marked as done.
+     *
+     * @param task Task that is marked as done.
+     * @return Task that is marked as done.
+     */
+    public String showDone(Task task) {
+        String done = String.format("Nice! I've marked this task as done:\n%s", task.toString());
+        return done;
+    }
+
+    /**
+     * Returns list of tasks that are stored.
+     *
+     * @param taskList The list of tasks that are stored.
+     * @return List of tasks that are stored.
+     * @throws DukeException If task list has error.
+     */
+    public String showList(TaskList taskList) throws DukeException {
+        int numTask = taskList.totalTask();
+        String task;
+
+        if (numTask == 0) {
+            task = "";
+        } else {
+            task = "1. " + taskList.getTask(1).toString();
+            for (int taskNumber = 2; taskNumber <= numTask; taskNumber++) {
+                task = task + "\n" + taskNumber + ". " + taskList.getTask(taskNumber).toString();
+            }
+        }
+
+        String listStatement = "Here are the tasks in your list:\n";
+        String output = listStatement + task;
+        return output;
+    }
+
+    /**
+     * Tells user that Duke can't understand the input.
+     *
+     * @return Lost message that tells user that Duke can't understand the input.
+     */
+    public String showLost() {
```

Comment:
Maybe simply return the string instead of creating an unnecessary variable? Saves a line of code 😄